app/jobs/regular/sync_topic_user_bookmarked.rb

```diff
@@ -5,35 +5,37 @@ class SyncTopicUserBookmarked < ::Jobs::Base
     def execute(args = {})
       raise Discourse::InvalidParameters.new(:topic_id) unless args[:topic_id].present?
 
-      DB.exec(<<~SQL, topic_id: args[:topic_id])
-        SELECT bookmarks.user_id, COUNT(*)
-        INTO TEMP TABLE tmp_sync_topic_user_bookmarks
+      bookmarks_exist = DB.exec(<<~SQL, topic_id: args[:topic_id]) > 0
+        SELECT 1
         FROM bookmarks
         LEFT JOIN posts ON posts.id = bookmarks.bookmarkable_id AND bookmarks.bookmarkable_type = 'Post'
-        LEFT JOIN topics ON (topics.id = bookmarks.bookmarkable_id AND bookmarks.bookmarkable_type = 'Topic') OR
-         (topics.id = posts.topic_id)
+        LEFT JOIN topics ON topics.id = bookmarks.bookmarkable_id AND bookmarks.bookmarkable_type = 'Topic'
         WHERE (topics.id = :topic_id OR posts.topic_id = :topic_id)
-        AND posts.deleted_at IS NULL AND topics.deleted_at IS NULL
-        GROUP BY bookmarks.user_id;
+        AND posts.deleted_at IS NULL
+        AND topics.deleted_at IS NULL
+        LIMIT 1
+      SQL
 
-        UPDATE topic_users
-        SET bookmarked = true
-        FROM tmp_sync_topic_user_bookmarks
-        WHERE topic_users.user_id = tmp_sync_topic_user_bookmarks.user_id AND
-          topic_users.topic_id = :topic_id AND
-          tmp_sync_topic_user_bookmarks.count > 0;
+      return unless bookmarks_exist
```

Comment:
```suggestion
      return if !bookmarks_exist
```

This is a Sam special :P